src/ngraph/runtime/gpu/op/lstm.cpp

```diff
@@ -0,0 +1,185 @@
+        {
+            throw ngraph_error("Incorrect number of new arguments");
+        }
+        return make_shared<Lstm>(new_args.at(0),
+                                 new_args.at(1),
+                                 new_args.at(2),
+                                 new_args.at(3),
+                                 new_args.at(4),
+                                 new_args.at(5),
+                                 new_args.at(6));
+    }
+    else
+    {
+        if (new_args.size() != 5 && m_fused_inputs)
+        {
+            throw ngraph_error("Incorrect number of new arguments");
+        }
+        return make_shared<Lstm>(
+            new_args.at(0), new_args.at(1), new_args.at(2), new_args.at(3), new_args.at(4));
+    }
+}
+
+op::gpu::Lstm::Lstm(std::shared_ptr<Node> input_xt_1,
+                    std::shared_ptr<Node> i2h_weights,
+                    std::shared_ptr<Node> hidden_state_ht_1,
+                    std::shared_ptr<Node> h2h_weights,
+                    std::shared_ptr<Node> i2h_bias,
+                    std::shared_ptr<Node> h2h_bias,
+                    std::shared_ptr<Node> cell_state_ct_1)
+    : RequiresTensorViewArgs("Lstm",
+                             {input_xt_1,
+                              i2h_weights,
+                              hidden_state_ht_1,
+                              h2h_weights,
+                              i2h_bias,
+                              h2h_bias,
+                              cell_state_ct_1})
+    , m_output_tensor_shape(hidden_state_ht_1->get_shape())
+    , m_output_cell_shape(cell_state_ct_1->get_shape())
+    , m_num_timesteps(1)
+    , m_num_gates_per_cell(4)
+    , m_src_sequence_length(1)
+    , m_src_layer_feature_size(static_cast<int>(input_xt_1->get_shape()[1]))
+    , m_src_iter_feature_size(static_cast<int>(hidden_state_ht_1->get_shape()[1]))
+    , m_num_cell_states(2)
+    , m_direction(1)
+    , m_num_fused_layers(1)
+    , m_fused_inputs(false)
+{
+    if (input_xt_1->get_shape().size() != i2h_weights->get_shape().size())
+    {
+        throw ngraph_error("input_xt_1 and i2h weights size dont match");
+    }
+
+    if (hidden_state_ht_1->get_shape().size() != h2h_weights->get_shape().size())
+    {
+        throw ngraph_error("hidden_state_ht_1 and h2h weights size dont match");
+    }
+
+    if (input_xt_1->get_shape().size() == 2)
+    {
+        m_batch_size = static_cast<int>(input_xt_1->get_shape()[0]);
+    }
+    else
+    {
+        throw ngraph_error("input_xt_1 doesnt have a rank 2");
+    }
+
+    if (shape_size(input_xt_1->get_shape()) !=
+        m_src_sequence_length * m_batch_size * m_src_layer_feature_size)
+    {
+        throw ngraph_error("input_xt_1 size is not equal t*n*c");
+    }
+
+    if (i2h_bias->get_shape()[0] != i2h_weights->get_shape()[0] ||
+        h2h_bias->get_shape()[0] != h2h_weights->get_shape()[0])
+    {
+        throw ngraph_error("bias and weights_shape are not compatible");
+    }
+
```

Comment:
Can you also add a check , to verify if the feature_size/shape of ht and ct are same?.